signer/src/storage/mod.rs

```diff
@@ -464,4 +477,28 @@ pub trait DbWrite {
         &self,
         prevout: &model::TxPrevout,
     ) -> impl Future<Output = Result<(), Error>> + Send;
+
+    /// Write the bitcoin tx sighash to the database.
+    fn write_bitcoin_tx_sighash(
+        &self,
+        sighash: &model::BitcoinTxSigHash,
+    ) -> impl Future<Output = Result<(), Error>> + Send;
+
+    /// Write the bitcoin transactions sighaes to the database.
+    fn write_bitcoin_txs_sighashes(
```

Comment:
I would probably just have the write-multiple versions of these methods, and take a slice instead of vec. Then if you just want to write one item then you just do `&[item]`.

Reply:
soved in https://github.com/stacks-network/sbtc/pull/877/commits/d0d00ff4170e84eb67a207b1d71663e901da911d